### PR TITLE
Add native functions to benchmarks where they exist

### PR DESCRIPTION
**Pull request by bjdixon**
- Added benchmarks for either (addresses #1913)
 - Removed isSet benchmarks as isSet has been removed from ramda
 - Changed indexOf to includes in the contains benchmark as includes is
semantically closer to contains than indexOf

### CODE REVIEW

**Comment by bjdixon** [comment]
I'm not sure that the additions I've made in (filter|find|findIndex)-where.bench.js are helpful as they mix native and ramda functions. In terms of benchmarking in general I would have thought it better to have a separate where.bench.js.

I'd like to create further PRs with benchmarks for the remaining unbenchmarked ramda functions (with native comparisons where appropriate) but will wait on feedback from this PR so I better understand the direction these should take.

File: lib/bench/either.bench.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+var R = require('../..');
+
+var gt10 = x => x > 10;
+var even = x => x % 2 === 0;
+
+module.exports = {
+  name: 'either',
+  tests: {
+    'either(gt10, even, 101)': function() {
+      R.either(gt10, even, 101)
```
Comment (kedashoe) [inline lib/bench/either.bench.js]: [Either](https://github.com/ramda/ramda/blob/v0.22.1/src/either.js) has an arity of 2 and returns a function so calling it like this doesn't do what it looks like.

File: lib/bench/contains.bench.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ module.exports = {
     'contains(c, cs)': function() {
       return contains(randomLowerAlpha(), alphabet);
     },
-    'native indexOf': function() {
-      return nativeIndexOf(randomLowerAlpha(), alphabet);
+    'native includes': function() {
```
Comment (kedashoe) [inline lib/bench/contains.bench.js]: I think it makes more sense to call this 'native contains' even if the native function we use is not called contains (either way definitely better than `indexOf` from before).

**Comment by kedashoe** [comment]
Thanks for the PR, the benchmark stuff could definitely use some :heart:

> I'm not sure that the additions I've made in (filter|find|findIndex)-where.bench.js are helpful as they mix native and ramda functions. In terms of benchmarking in general I would have thought it better to have a separate where.bench.js.

Agreed, not sure we need native comparisons in benchmarks that test more than one ramda function.

In general I think it would be nice to have separate `Ramda vs Ramda` and `Ramda vs native vs etc` suites. Ramda vs others is interesting every now and then, but usually I just care if the changes I'm making make ramda itself faster or slower. Would also be nice to auto-compare a branch vs master.

File: lib/bench/either.bench.js (hunk 1 above)
Comment (bjdixon) [inline lib/bench/either.bench.js]: :flushed: oops

**Comment by buzzdecafe** [comment]
> In general I think it would be nice to have separate `Ramda vs Ramda` and `Ramda vs native vs etc`

:+1: for this idea -- although it implies we would track benches over time?

**Comment by CrossEye** [comment]
Wow, I missed this one entirely!  Very nice.

**Pull request merged**